NCR-Corporation/sample-app-burgers
# MARKET 3997: Implement mini-profiler tool (POC)

## Description
This pull request implements a mini-profiler tool (POC). From the ticket description on Jira:
> Log requests to the platform from the sample app and display it to the user in a miniprofiler-like way. We need to strip the credentials that are used for the request as to not show them to the user. No CX, copy miniprofiler.
> 
> Filter what's reported by the profiler to only show the calls to the platform. Mask / obfuscate credentials.

## Screenshots/Screencasts
![image](https://user-images.githubusercontent.com/68287936/98303774-b192c300-1f8c-11eb-9944-f6644b3ff5e6.png)
![image](https://user-images.githubusercontent.com/68287936/98303842-d9822680-1f8c-11eb-9de2-00f4e7a745b7.png)
![image](https://user-images.githubusercontent.com/68287936/98303935-00405d00-1f8d-11eb-97dc-a6c3d6175500.png)


## Testing
1. Run the project locally and open the app in your browser.
2. Click the 'Profiler' tab to expand the debug toolbar.
3. Review the information in each toolbar item ('Profiling', 'Requests', etc.).
4. Check that all user credentials/sensitive info is hidden in 'Settings' tab.
4. Repeat for all views/pages/urls within the app site.


## Ticket
[MARKET-3997](https://jira.ncr.com/browse/MARKET-3997)
